### PR TITLE
php2branch > tl1revbranch

### PR DESCRIPTION
**Pull request by php-2**
Tl1revbranch > php2branch

### CODE REVIEW

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
 include_once DOL_DOCUMENT_ROOT.'/custom/constants.php';
 include_once DOL_DOCUMENT_ROOT.'/custom/extintegration/class/clsHost.class.php';
-  $response = array();
-    if($_SERVER['REQUEST_METHOD']=='POST')
+if (__DEBUG__) {echo '<br>if post request method</br>';}
+$host   = $apiPar[0];
+$parameters = array("apiPar0" => "host");
+$fileds = array(  "fld0" => "check_result");
+$help   = array("parameters"=>$parameters,"fileds"=>$fileds);
+if ($apiHelp)
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php]: where this parameter $apiHelp defined ? how you fill its value?

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php [hunk 2]
```diff
@@ -1,30 +1,47 @@
 <?php
-include_once '../../../master.inc.php';
 include_once DOL_DOCUMENT_ROOT.'/custom/constants.php';
 include_once DOL_DOCUMENT_ROOT.'/custom/extintegration/class/clsHost.class.php';
-  $response = array();
-    if($_SERVER['REQUEST_METHOD']=='POST')
+if (__DEBUG__) {echo '<br>if post request method</br>';}
+$host   = $apiPar[0];
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php]: where this parameter $apiPar[0] defined ? how you fill its value?

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php [hunk 3]
```diff
@@ -1,30 +1,47 @@
...
+            $response["data"] = S_NULL;
         }
-    echo json_encode($response);
+   }
+   else
+    {
+        $response["error"] = S_TRUE;
+        $response["errid"] = "ERR_NO_PAR_VALUES";
+    }
+}
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php]: how you path the $response back to wrapper ?

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/user_login/apiuserlogin.php [hunk 4]
```diff
@@ -1,61 +1,53 @@
...
-            $response["errid"] = "ERR_NO_PAR_VALUES";
-       }
+   }
+    else
+    {
+        $response["error"] = S_TRUE;
+        $response["errid"] = "ERR_NO_PAR_VALUES";
     }
-
+}
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/user_login/apiuserlogin.php]: same comments exactly as the previous apicheckhost.php

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php (hunk 1 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php]: defined in the apiwrapper.php and it's value passed as parameter called apiHelp

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php (hunk 2 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php]: defined in the apiwrapper.php and it's value passed as parameter called apiPar0

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php (hunk 3 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/extintegration/class/apicheckhost.php]: because encoding response to json is in wrapper

**Pull request merged**